app/scripts/controllers/encryption-public-key.ts

```diff
@@ -0,0 +1,443 @@
+   * @param [req] - The original request, containing the origin.
+   */
+  async newRequestEncryptionPublicKey(
+    address: string,
+    req: OriginalRequest,
+  ): Promise<string> {
+    const keyring = await this._keyringController.getKeyringForAccount(address);
+
+    switch (keyring.type) {
+      case HardwareKeyringTypes.ledger: {
+        return new Promise((_, reject) => {
+          reject(
+            new Error('Ledger does not support eth_getEncryptionPublicKey.'),
+          );
+        });
+      }
+
+      case HardwareKeyringTypes.trezor: {
+        return new Promise((_, reject) => {
+          reject(
+            new Error('Trezor does not support eth_getEncryptionPublicKey.'),
+          );
+        });
+      }
+
+      case HardwareKeyringTypes.lattice: {
+        return new Promise((_, reject) => {
+          reject(
+            new Error('Lattice does not support eth_getEncryptionPublicKey.'),
+          );
+        });
+      }
+
+      case HardwareKeyringTypes.qr: {
+        return Promise.reject(
+          new Error('QR hardware does not support eth_getEncryptionPublicKey.'),
+        );
+      }
+
+      default: {
+        return this._encryptionPublicKeyManager.addUnapprovedMessageAsync(
+          { from: address },
+          req,
+        );
+      }
+    }
+  }
+
+  /**
+   * Signifies a user's approval to receiving encryption public key in queue.
+   *
+   * @param msgParams - The params of the message to receive & return to the Dapp.
+   * @returns A full state update.
+   */
+  async encryptionPublicKey(msgParams: EncryptionPublicKeyParamsMetamask) {
+    log.info('MetaMaskController - encryptionPublicKey');
+    const messageId = msgParams.metamaskId as string;
+    // sets the status op the message to 'approved'
+    // and removes the metamaskId for decryption
+    try {
+      const cleanMessageParams =
+        await this._encryptionPublicKeyManager.approveMessage(msgParams);
+
+      // EncryptionPublicKey message
+      const publicKey = await this._keyringController.getEncryptionPublicKey(
+        cleanMessageParams.from,
+      );
+
+      // tells the listener that the message has been processed
+      // and can be returned to the dapp
+      this._encryptionPublicKeyManager.setMessageStatusReceived(
+        messageId,
+        publicKey,
+      );
+
+      this._acceptApproval(messageId);
+
+      return this._getState();
+    } catch (error) {
+      log.info(
+        'MetaMaskController - eth_getEncryptionPublicKey failed.',
+        error,
+      );
+      this._cancelAbstractMessage(this._encryptionPublicKeyManager, messageId);
+      throw error;
+    }
+  }
+
+  /**
+   * Used to cancel a message submitted via eth_getEncryptionPublicKey.
+   *
+   * @param msgId - The id of the message to cancel.
+   */
+  cancelEncryptionPublicKey(msgId: string) {
+    this._cancelAbstractMessage(this._encryptionPublicKeyManager, msgId);
```

Comment:
As there's nothing else here and only used once, probably simplest to move the body of `_cancelAbstractMessage` here.

Reply:
`_cancelAbstractMessage` is used in a few places (`encryptionPublicKey`, `cancelEncryptionPublicKey` and `rejectUnapproved`).